Rakefile.rb

```diff
@@ -48,6 +48,36 @@ def name_from(json)
   orgs.first[:name]
 end
 
+def file_to_commit_metadata
+  # There's about 2MB of data returned by this command, so rather than
+  # using backticks to turning it into a string and then splitting it,
+  # parse the output line-by-line. As a bonus this doesn't
+  # unnecessarily invoke a shell.
+  command = ['git', '--no-pager', 'log', '--name-only',  '--format=%h|%at', '--', 'data']
+  result = {}
+  sha, timestamp = nil
+  Kernel.open('|-', 'r') { |f|
+    if f
+      # In the parent process after forking, parse the output:
+      f.each_line { |line|
+        commit_match = line.match /^([a-f\d]+)\|(\d+)$/
+        if commit_match
+          sha, timestamp = commit_match.captures
+        else
+          filename = line.strip
+          unless result.has_key? filename
+            result[filename] = {sha: sha, timestamp: timestamp}
+          end
```

Comment:
I've dealt with those comments in 5fa50f26277f42b19a hopefully - to mitigate it being less clear what `line` means, I added a comment about the format of the data. Also I noticed that the empty string is being uselessly set for each blank line, so that commit also includes a change to not process blank llines.